### PR TITLE
Implement transitions for bullet charts

### PR DESCRIPTION
**Pull request by acasaccia**
Add transitions handling on bullet and bulletChart models and adapt the examples as a result.

### CODE REVIEW

**Comment by liquidpele** [comment]
This PR contains build files, please remove all build/* files from the PR.

**Comment by acasaccia** [comment]
@liquidpele build files have been removed. The chart seems to be working properly, however tests are failing because of timing issues related to animations.

File: src/models/bullet.js [hunk 1]
```diff
@@ -276,6 +285,7 @@ nv.models.bullet = function() {
         width:    {get: function(){return width;}, set: function(_){width=_;}},
         height:    {get: function(){return height;}, set: function(_){height=_;}},
         tickFormat:    {get: function(){return tickFormat;}, set: function(_){tickFormat=_;}},
+        bulletDuration:    {get: function(){return bulletDuration;}, set: function(_){bulletDuration=_;}},
```
Comment (liquidpele) [inline src/models/bullet.js]: This is just called duration on other charts, so it should match.  

See the "duration" option on most charts:   http://nvd3-community.github.io/nvd3/examples/documentation.html

File: src/models/bulletChart.js [hunk 2]
```diff
@@ -21,9 +21,10 @@ nv.models.bulletChart = function() {
         , width = null
         , height = 55
         , tickFormat = null
-	, ticks = null
+	    , ticks = null
```
Comment (liquidpele) [inline src/models/bulletChart.js]: There appears to be a tab there messing things up, should be all spaces to match the rest of the code.

File: src/models/bulletChart.js [hunk 3]
```diff
@@ -189,6 +194,7 @@ nv.models.bulletChart = function() {
         tickFormat:    {get: function(){return tickFormat;}, set: function(_){tickFormat=_;}},
         ticks:    {get: function(){return ticks;}, set: function(_){ticks=_;}},
         noData:    {get: function(){return noData;}, set: function(_){noData=_;}},
+        axisDuration:    {get: function(){return axisDuration;}, set: function(_){axisDuration=_;}},
```
Comment (liquidpele) [inline src/models/bulletChart.js]: This appears to be a duration just for the axis, so this name seems fine, but please document any new options you add in the json at the bottom of documentation.html

**Comment by liquidpele** [comment]
Also, github says there are conflicts here so it can't merge, you'll want to merge in latest master to your branch and fix any conflicts there.   

Also the tests need to be passing... maybe changed to be more valid.

File: src/models/bulletChart.js (hunk 3 above)
Comment (acasaccia) [inline src/models/bulletChart.js]: I just removed `axisDuration` and found a way to reuse the underlying bullet model `duration`. This way one can control both the bullet and axis duration with the standard `duration` parameter

File: src/models/bulletChart.js [hunk 4]
```diff
@@ -127,6 +127,8 @@ nv.models.bulletChart = function() {
 
             // Transition the updating ticks to the new scale, x1.
             var tickUpdate = d3.transition(tick)
+                .transition()
+                .duration(bullet._options.duration)
```
Comment (liquidpele) [inline src/models/bulletChart.js]: Should do bullet.duration() to get that duration value.

**Comment by liquidpele** [comment]
Still has failing tests

**Comment by acasaccia** [comment]
5057ed7 seems to have fixed the tests relative to bullet chart.
The latest build failure seems not related to my changes...
May it be that the tests on MultiBar Chart are unstable for unrelated reasons?

**Comment by liquidpele** [comment]
Okay, I think this looks good.   Not sure what's up with that test under multibar but as you said that should be unrelated.

**Pull request merged**